vars/deployPlannerSnapshot.groovy

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/groovy
+
+import java.util.regex.Pattern
+
+def call(body) {
+    // evaluate the body block, and collect configuration into the object
+    def config = [:]
+    body.resolveStrategy = Closure.DELEGATE_FIRST
+    body.delegate = config
+    body()
+
+    def yaml
+    def openShiftProject = config.openShiftProject
+    def openShiftTemplate = config.openShiftTemplate
+    def originalImageName = config.originalImageName
+    def newImageName = config.newImageName
+    def deploymentName = config.githubRepo
+    def providerLabel = config.providerLabel ?: 'fabric8'
+    def project = config.githubProject
+
+    def flow = new io.fabric8.Fabric8Commands()
+    def utils = new io.fabric8.Utils()
+
+    openShiftProject = openShiftProject + '-' + utils.getRepoName()
```

Comment:
IMHO, keeping variables as immutable as possible makes it easy to reason with the code. Here openshiftProject isn't acting like an accumulator (sum, product etc...) so we can keep it immutable like mentioned in the comment above

Reply:
:+1: